### PR TITLE
feat(backend-mavericks): Implement CRUD operation for "Tag" Feature

### PR DESCRIPTION
**Pull request by Edilcc**
Implement Tag API endpoint using RESTful design principles.

Create endpoint that accept a JSON payload containing the tag data, and return the newly created tag object.

Read all endpoint that return an array of all tag objects.

Read blog-tag endpoint, it return an array of all tag objects associated to a blog.

Update endpoint, it accept a tag ID as a parameter, and a JSON payload containing the updated tag data, and return the updated tag object.

Delete endpoint that would accept a tag ID as a parameter, delete the corresponding tag object, and return a success message.

All endpoints have  return appropriate HTTP status codes and error messages for invalid requests.

Implement API that tested thoroughly to ensure all endpoints function correctly and as expected.

### CODE REVIEW

File: Backend/starter_project_mavericks/BlogApp/BlogApp.Application/Features/_Tags/DTOs/Create_TagDto.cs
Comment (TigistShiferaw) [inline Backend/starter_project_mavericks/BlogApp/BlogApp.Application/Features/_Tags/DTOs/Create_TagDto.cs:1]: Use Camel Case variable and file naming
e.g  CreateTagDto

File: Backend/starter_project_mavericks/BlogApp/BlogApp.Application/Features/_Tags/CQRS/Handlers/Get_TagDetailQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        private readonly IMapper _mapper;
+
+        public Get_TagDetailQueryHandler(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+        public async Task<_TagDto> Handle(Get_TagDetailQuery request, CancellationToken cancellationToken)
+        {
+            var _Tag = await _unitOfWork._TagRepository.Get(request.Id);
```
Comment (TigistShiferaw) [inline Backend/starter_project_mavericks/BlogApp/BlogApp.Application/Features/_Tags/CQRS/Handlers/Get_TagDetailQueryHandler.cs:26]: You should handle not found case like this

if (_Tag == null)
            {
                var error = new NotFoundException(nameof(_Tag), request.Id);
                return new BaseResponse<Unit>
                {
                    Success = false,
                    Message = "Tag deletion failed",
                    Errors = new List<string> { error.Message }
                };
            }

File: Backend/starter_project_mavericks/BlogApp/BlogApp.Application/Features/_Tags/DTOs/Create_TagDto.cs
Comment (Edilcc) [reply Backend/starter_project_mavericks/BlogApp/BlogApp.Application/Features/_Tags/DTOs/Create_TagDto.cs:1]: Resolved

File: Backend/starter_project_mavericks/BlogApp/BlogApp.Application/Features/_Tags/CQRS/Handlers/Get_TagListQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        public Get_TagListQueryHandler(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task<List<_TagDto>> Handle(Get_TagListQuery request, CancellationToken cancellationToken)
+        {
+            var _Tags = await _unitOfWork._TagRepository.GetAll();
+            Console.Write(_Tags);
```
[the remark this answers is not in the document]
Comment (Edilcc) [reply Backend/starter_project_mavericks/BlogApp/BlogApp.Application/Features/_Tags/CQRS/Handlers/Get_TagListQueryHandler.cs]: Resolved

**Pull request merged**